### ISSUE TITLE
sign crashes if key file contains trailing newline

### ISSUE THREAD

**Issue by frigus02** [wasmCloud/wash-archive#66]
Hi there. I just started playing with wasmCloud and it's going great so far. I started generating my own key files before realizing `wash` does this automatically if I don't have any, yet.

Anyway, my files ended with a trailing newline and that caused `wash claims sign` to crash with a syack overflow message:

```
$ wash claims sign ...
thread 'main' has overflowed its stack
fatal runtime error: stack overflow
```

You should be able to reproduce like this:

```
$ wash keys gen account -o json | jq '.seed' -r > account.nk
$ wash claims sign --issuer account.nk path-to-a-wasm-file.wasm --name dummy
```

What would be the correct behaviour for key files with a trailing newline?

**Comment by brooksmtownsend** [comment]
Thanks for the issue @frigus02 🙂 This was actually a bug with the `nkeys` crate and merged into the main branch yesterday https://github.com/wasmCloud/nkeys/pull/8, so this will be patched very soon with `wash`. It will just take a version update, if you're compiling from source locally you can try updating your nkeys version in `Cargo.toml`:
```
nkeys = "0.0.12"
```

Regardless, I'll open up a PR for this today so we can get this fixed in the `main` branch

**Comment by brooksmtownsend** [comment]
To answer your question directly, key files with a trailing newline should not cause a stackoverflow 😄

**Comment by frigus02** [comment]
Ha, nice. I did actually look at the `nkeys` repository but didn't see any open issues. Thanks for the quick response 👍

**Comment by brooksmtownsend** [comment]
`wash` also depends on `wascap` and `provider-archive` which will need to be updated to the newest versions too, I'll tag this issue once I get a PR in for `wash` but it appears that you'll need to hold on updating the dependency if you built from source. For now I'd just recommend checking your keyfiles for newlines, thanks for your patience @frigus02

**Comment by frigus02** [comment]
No need to hurry. The files that `wash` creates itself are without newlines anyway. So that's not a big issue.

**Comment by brooksmtownsend** [comment]
Fixed by #69 , I can't reproduce after the `nkeys` fix was put into place. Let me know if you run into any other issues @frigus02 !

**Comment by frigus02** [comment]
Confirmed. Works fine with and without trailing newline in v0.1.16 🎉
